### ISSUE TITLE
Timescape adding <span> elements to top-level of DOM

### ISSUE THREAD

**Issue by Kay2dan** [dan-lee/timescape#11]
Hi Dan,

version: 0.4.0
library: React
browsers tested: Brave (latest), Firefox (latest)

Issue: 
Timescape is adding <span> elements to the top level in DOM.
Timescape keeps inserting new <span> elements every time a component containing the Timescape is rendered. 

See attached screenshot. 

**My implementation:**

```jsx
function LabeledDateTimeField({
  name,
  label,
  dateTime,
  isDisabled,
}: {
  name: string;
  label: string;
  dateTime?: Date;
  isDisabled: boolean;
}) {
  const { register, setValue } = useFormContext();
  const { getRootProps, getInputProps } = useTimescape({
    date: dateTime,
    digits: "2-digit",
    onChangeDate: nextDate => {
      setValue(name, nextDate);
    },
  });

  return (
    <>
      <div className={`flex flex-col justify-start my-2`}>
        <span className="text-xs text-gray-4">{`${label}: `}</span>

        <input {...register(label)} hidden />

        <div
          className="relative flex items-center gap-[1px] w-fit border-b border-gray-3 px-1 select-none focus-within:outline focus-within:outline-1"
          {...getRootProps()}>
          <input
            className="h-fit max-w-[4ch] font-inherit text-inherit leading-[inherit] tabular-nums border-none bg-transparent caret-transparent focus:bg-independence focus:text-eggShell-1"
            {...getInputProps("years")}
            disabled={isDisabled}
            style={{ fontFamily: "inherit", fontSize: "inherit" }}
          />
          <span className="text-gray-4">/</span>
          <input
            className="h-fit max-w-[2ch] font-inherit text-inherit leading-[inherit] tabular-nums border-none bg-transparent caret-transparent focus:bg-independence focus:text-eggShell-1"
            {...getInputProps("months")}
            disabled={isDisabled}
            style={{ fontFamily: "inherit", fontSize: "inherit" }}
          />
          <span className="text-gray-4">/</span>
          <input
            className="h-fit max-w-[2ch] font-inherit text-inherit leading-[inherit] tabular-nums border-none bg-transparent caret-transparent focus:bg-independence focus:text-eggShell-1"
            {...getInputProps("days")}
            disabled={isDisabled}
            style={{ fontFamily: "inherit", fontSize: "inherit" }}
          />
          <span className="text-gray-4"> </span>
          <input
            className="h-fit max-w-[2ch] font-inherit text-inherit leading-[inherit] tabular-nums border-none ml-1 bg-transparent caret-transparent focus:bg-independence focus:text-eggShell-1"
            {...getInputProps("hours")}
            disabled={isDisabled}
            style={{ fontFamily: "inherit", fontSize: "inherit", fontWeight: "bold" }}
          />
          <span className="font-bold text-gray">:</span>
          <input
            className="h-fit max-w-[2ch] font-inherit text-inherit leading-[inherit] tabular-nums border-none bg-transparent caret-transparent focus:bg-independence focus:text-eggShell-1"
            {...getInputProps("minutes")}
            disabled={isDisabled}
            style={{ fontFamily: "inherit", fontSize: "inherit", fontWeight: "bold" }}
          />
          <span className="text-gray-4">min</span>
        </div>
      </div>
    </>
  );
}

```

**Screenshot from Elements tab within Brave's devtools:**

![Screenshot from 2024-03-05 12-42-02](https://github.com/dan-lee/timescape/assets/6276091/9b30ec0f-25d9-48c2-bdbe-7bcdf32a0b20)

**Comment by dan-lee** [comment]
Hey @Kay2dan, thanks for the bug report. It looks like this is running in Next.js? Just to know so I'll be able to reproduce this

**Comment by Kay2dan** [comment]
> Hey @Kay2dan, thanks for the bug report. It looks like this is running in Next.js? Just to know so I'll be able to reproduce this

Yes it is. Im using Blitz.js which is based on Next.js.

Blitz.js: https://blitzjs.com/
Using Next.js: 13.4

Thanks for looking into this

**Comment by dan-lee** [comment]
I've released a fix in [v0.4.1](https://github.com/dan-lee/timescape/releases/tag/timescape%400.4.1). Can you check if it works correctly for you now?

I also figured that SSR stumbles with missing `window`, so I've added a check for `ResizeObserver` and `MutationObserver` in case you worked around that somehow.

**Comment by Kay2dan** [comment]
> I've released a fix in [v0.4.1](https://github.com/dan-lee/timescape/releases/tag/timescape%400.4.1). Can you check if it works correctly for you now?
> 
> I also figured that SSR stumbles with missing `window`, so I've added a check for `ResizeObserver` and `MutationObserver` in case you worked around that somehow.

Thank you. Will update & check shortly & get back to you.
